src/drt/src/io/GuideProcessor.cpp

```diff
@@ -369,80 +369,270 @@ void connectGuidesWithBestPinLoc(const Point3D& guide_pt,
-      }
-      frLayerNum layerNum = layer->getLayerNum();
+  }
+  return false;
+}
+/**
+ * @brief Adds Bridge guides for touching guides on the same layer if needed.
+ *
+ * The function identifies touching guides on the same layer on consecutive
+ * track indices. It adds a bridge guide on the upper or lower layer that
+ * connects both guides if such one does not exist.
+ */
+void addTouchingGuidesBridges(TrackIntervalsByLayer& intvs, utl::Logger* logger)
+{
+  struct BridgeGuide
+  {
+    frCoord track_idx{-1};
+    frCoord begin_idx{-1};
+    frCoord end_idx{-1};
+    frLayerNum layer_num{-1};
+    BridgeGuide(frCoord track_idx_in,
+                frCoord begin_idx_in,
+                frCoord end_idx_in,
+                frLayerNum layer_num_in)
+        : track_idx(track_idx_in),
+          begin_idx(begin_idx_in),
+          end_idx(end_idx_in),
+          layer_num(layer_num_in)
+    {
+    }
+  };
 
-      // get the top layer for a pin of the net
-      bool isAboveTopLayer = false;
-      for (const auto& bterm : net->getBTerms()) {
-        isAboveTopLayer = bterm->isAboveTopLayer();
+  std::vector<BridgeGuide> bridge_guides;
+  // append touching edges
+  for (int layer_num = 0; layer_num < (int) intvs.size(); layer_num++) {
+    const auto& curr_layer_intvs = intvs[layer_num];
+    int prev_track_idx = -2;
+    for (const auto& [curr_track_idx, curr_track_intvs] : curr_layer_intvs) {
+      if (curr_track_idx == prev_track_idx + 1) {
+        const auto& prev_track_intvs = curr_layer_intvs.at(prev_track_idx);
+        const auto intvs_intersection = curr_track_intvs & prev_track_intvs;
+        for (const auto& intv : intvs_intersection) {
+          const auto begin_idx = intv.lower();
+          const auto end_idx = intv.upper();
+          bool has_bridge = false;
+          // lower layer intersection
+          if (layer_num - 2 >= 0) {
+            has_bridge = hasGuideInterval(begin_idx,
+                                          end_idx,
+                                          curr_track_idx,
+                                          prev_track_idx,
+                                          layer_num - 2,
+                                          intvs);
+          }
+          if (layer_num + 2 < (int) intvs.size() && !has_bridge) {
+            has_bridge = hasGuideInterval(begin_idx,
+                                          end_idx,
+                                          curr_track_idx,
+                                          prev_track_idx,
+                                          layer_num + 2,
+                                          intvs);
+          }
+          if (!has_bridge) {
+            // add bridge guide;
+            frLayerNum bridge_layer_num;
+            if (layer_num + 2 < (int) intvs.size()) {
+              bridge_layer_num = layer_num + 2;
+            } else if (layer_num - 2 >= 0) {
+              bridge_layer_num = layer_num - 2;
+            } else {
```

Comment:
You could set bridge_layer_num above rather than repeating the conditions (std::optional to keep track if its set).